### DISCUSSION TITLE
What is the best way to load video to PIXI.js v7?

### DISCUSSION THREAD

**Discussion by MannySVK** [pixijs/pixijs#9212] [Q&A] [answered]
Hello, what is the best to load a video in PIXI.js v7? Is there any way how to also show loading progress?

**Comment by SuperSodaSea** [comment, marked as answer]
Here is a similar discussion: <https://github.com/pixijs/pixijs/discussions/9083#discussioncomment-4729311>

> ```js
> const video = document.createElement('video'); // also could be path/to/video.mp4
> const texture = Texture.from(video);
> ```

Then you can use `texture.baseTexture.resource.source` to get the underlying `HTMLVideoElement`, and you can do whatever you want with it (e.g. `.play()`, `.pause()`). [This StackOverflow question](https://stackoverflow.com/questions/5029519/html5-video-percentage-loaded) has some information about how to get the loading progress of a `HTMLVideoElement`.
